ethcore/src/miner/miner.rs

```diff
@@ -1412,22 +1416,20 @@ impl miner::MinerService for Miner {
 				});
 		}
 
-		if has_new_best_block || (imported.len() > 0 && self.options.reseal_on_uncle) {
+		if has_new_best_block {
 			// Reset `next_allowed_reseal` in case a block is imported.
 			// Even if min_period is high, we will always attempt to create
 			// new pending block.
 			self.sealing.lock().next_allowed_reseal = Instant::now();
 
-			if !is_internal_import {
+			if !is_internal_import && imported.len() > 0 && self.options.reseal_on_uncle {
```

Comment:
Perhaps this stuff is not needed? Seems that my patch might have been made on outdated version of master.